### PR TITLE
hilt android tutorial

### PR DESCRIPTION
**Pull request by RaphaelNdonga**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by RaphaelNdonga** [comment]
This fixes #2050

File: articles/hilt-android-tutorial/index.md [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+There are a few things to note in `MainActivity`.
+
+1. `MainActivity` initializes its dependencies. Therefore, its dependencies are only available during its lifetime. This makes `MainActivity` as a *component*.
+
+2. `MainActivity` has gets the dependencies. This makes it a *dependency container*.
+
+## Dependency injection with hilt
+Manual dependency injection works . However, as the app scales, it becomes cumbersome to manage dependencies. Hilt has a higher cost of set up. But it proves to be very beneficial to an app that scales.
+
+## Getting Started with Hilt
```
Comment (WanjaMIKE) [inline articles/hilt-android-tutorial/index.md:112]: Could you use ```###``` rather than ```##```

File: articles/hilt-android-tutorial/index.md [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+2. Dependency Injection with hilt.
+3. Hilt and Interfaces.
+4. Hilt and 3rd Party Libraries
+5. Conclusion
+## Getting Started
+If you understand git, you can access this project from the following link: [Hilt Tutorial](https://github.com/RaphaelNdonga/hilt-tutorial).  
+Each step has its own branch.
+
+Otherwise, create a new android studio project.
+## What is dependency injection?
```
Comment (WanjaMIKE) [inline articles/hilt-android-tutorial/index.md:23]: Combine ```## What is dependency injection?``` section with the introduction

File: articles/hilt-android-tutorial/index.md [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+}
+```
+
+Create a new class that extends `Application()` and annotate it as follows:
+```kotlin
+@HiltAndroidApp
+class MyApplication:Application() {
+}
+```
+This gives hilt access to the entire application. It creates a *dependency container* at the application level. Hilt can supply dependencies to any part of the app.
```
Comment (WanjaMIKE) [inline articles/hilt-android-tutorial/index.md:147]: Replace `hilt` with `Hilt` across the article

File: articles/hilt-android-tutorial/index.md [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+```
+
+Run your app and open the logcat. Search for 'SpanishPerson':
+```
+com.example.android.hilttutorial I/SpanishPerson: Despacito senor
+```
+
+The code works now.
+
+## Part 4: Hilt and 3rd Party Libraries
```
Comment (WanjaMIKE) [inline articles/hilt-android-tutorial/index.md:395]: Replace `3rd` with `third`

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
Nice article @RaphaelNdonga. Could you review the above comments

**Comment by RaphaelNdonga** [comment]
@WanjaMIKE  I made the changes you requested

**Review by WanjaMIKE — APPROVED** [review]
Nice tutorial. Easy to follow and understand. Keep up!

**Comment by WanjaMIKE** [comment]
As you know, we were working on a new structure for our EngEd Repo, which has led to a slight slow down of article publishing. Everything is almost ready now.
In order to publish your article as soon as possible, please do the following:

Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
Delete your article directory from the articles folders.
Do a git pull (take a look at the new structure of the repo after a git pull).
Upload your article files to the new articles folder inside content.
Ping me via Slack if you have any problems.```

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by RaphaelNdonga** [comment]
@hectorkambow @WanjaMIKE  did I perform the task correctly?

**Comment by hectorkambow** [comment]
> @hectorkambow @WanjaMIKE did I perform the task correctly?

@RaphaelNdonga It seems to be in the correct structure I'll know for sure when I check it out 👍 

Thank you

**Comment by hectorkambow** [comment]
great work gents 👍  Awesome tutorial - real informative and easy to follow along - keep up the great work 🚀

**Pull request merged**